src/autotype/AutoTypeAction.h

```diff
@@ -28,8 +28,61 @@ class AutoTypeExecutor;
 class KEEPASSXC_EXPORT AutoTypeAction
 {
 public:
+    class Result
```

Comment:
I feel like this should just be a `QPair<AutoTypeAction::ResultEnum, QString>` where the result enum is "Ok, Retry, Failed" and the QString contains the error message (optional). The below is a lot of boiler plate to provide the same end result.

Reply:
Kind of but it's not very ergonomic to use enums and QPairs and the code isn't self explanatory compared to the boilerplate version. 

IIRC I would need to return something like this:

```
return qMakePair(AutoTypeActionResult_Failed, tr("Well duh.);
```

It doesn't look very pretty to use a QPair as a return type either so it makes declarations more verbose as well. Also need to remember returning an Ok looks a bit weird as you need to create an empty string. The enum values need to be globally unique unless namespaced.

Reply:
Doesn't the enum value get wrapped in the class namespace? So it would be AutoTypeAction::Ok

I am not against the current version it is just verbose. We can keep it.

Reply:
Indeed, the enum would actually work like that but it would reserve the words inside the class. Not necessarily an issue.

Though as discussed on IRC and here I'll keep this implementation for now and we'll refactor later if we feel the urge to either use a more global result class or just get rid of it.